src/core/app.lua

```diff
@@ -65,6 +66,8 @@ end
 -- Run app:methodname() in protected mode (pcall). If it throws an
 -- error app will be marked as dead and restarted eventually.
 local function with_restart (app, method)
```

Comment:
I guess we should rename `with_restart` to something like `invoke_app_method`. Nothing critical but this change makes the functions name a misnomer.